pkg/plugins/policies/meshtimeout/plugin/v1alpha1/plugin.go

```diff
@@ -185,10 +185,14 @@ func applyToGateway(
 
 func configure(rules core_xds.Rules, subset core_xds.Subset, protocol core_mesh.Protocol, listener *envoy_listener.Listener, cluster *envoy_cluster.Cluster, routeActions []*envoy_route.RouteAction) error {
 	var conf api.Conf
-	if computed := rules.Compute(subset); computed != nil {
-		conf = computed.Conf.(api.Conf)
+	if rules == nil {
+		conf = api.Conf{}
 	} else {
-		return nil
+		if computed := rules.Compute(subset); computed != nil {
```

Comment:
I know this PR is already in master but this seems weird to me. `rules` is `[]*core_xds.Rule`. Why are we differentiating between the `nil` list and the empty list here?

Reply:
Can you elaborate? Are you talking about a situation when `rules` list is empty, so the `computed` is nil?

Reply:
I'll take another look in master, feel free to ignore but it's nothing to do with `computed`. This code is checking `rules == nil` but rules is a slice, right? Not a pointer. So it's taking two different paths based on whether we have a `nil` slice or an empty `[]*core_xds.Rule{}` slice. Which is generally unexpected in go, because it's not clear to me the semantics of an empty rules slice vs a `nil` rules slice.